src/emu-i386/simx86/cpatch.c

```diff
@@ -111,8 +111,9 @@ asmlinkage void rep_movs_stos(struct rep_stack *stack)
 	else if (*eip & 1)
 		size = 4;
 	len *= size;
-	m_munprotect(addr - ((EFLAGS & EFLAGS_DF) ? (len - size) : 0),
-		     len, eip);
+	if (!vga_write_access(addr))
+		m_munprotect(addr - ((EFLAGS & EFLAGS_DF) ? (len - size) : 0),
+			     len, eip);
```

Comment:
A comment would be good.
I guess this logic means "if !vga memory then
unprotect here, otherwise the vga access functions
will unprotect themselves", but I am not really sure.